File: loadercomplex-forge/src/ver182/java/com/enderzombi102/loadercomplex/forge18/impl/entity/ForgeLivingEntity.java
package com.enderzombi102.loadercomplex.forge18.impl.entity;

import com.enderzombi102.loadercomplex.api.minecraft.entity.LivingEntity;
import com.enderzombi102.loadercomplex.api.minecraft.entity.Player;
import com.enderzombi102.loadercomplex.api.minecraft.item.EquipmentSlot;
import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack;
import com.enderzombi102.loadercomplex.forge18.impl.item.ForgeItemStack;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.util.Hand;

public class ForgeLivingEntity extends ForgeEntity implements LivingEntity {
	private final net.minecraft.entity.LivingEntity backingEntity;

	public ForgeLivingEntity(net.minecraft.entity.LivingEntity backingEntity) {
		super(backingEntity);
		this.backingEntity = backingEntity;
	}

	@Override
	public float getHealth() {
		return this.backingEntity.getHealth();
	}

	@Override
	public void setHealth(float health) {
		this.backingEntity.setHealth( health );
	}

	@Override
	public boolean isPlayer() {
		return this.backingEntity instanceof PlayerEntity;
	}

	@Override
	public Player asPlayer() {
		return new ForgePlayer( (PlayerEntity) this.backingEntity );
	}

	@Override
	public boolean isChild() {
		return this.backingEntity.isBaby();
	}

	@Override
	public boolean canBreathUnderwater() {
		return this.backingEntity.canBreatheInWater();
	}

	@Override
	public boolean isAttachedToLadder() {
		return this.backingEntity.isClimbing();
	}

	@Override
	public int getArmorValue() {
		return this.backingEntity.getArmor();
	}

	@Override
	public ItemStack getItemInMainHand() {
		return new ForgeItemStack( this.backingEntity.getStackInHand(Hand.MAIN_HAND) );
	}

	@Override
	public ItemStack getItemInOffHand() {
		return new ForgeItemStack( this.backingEntity.getStackInHand(Hand.OFF_HAND) );
	}

	@Override
	public boolean hasItemInSlot(EquipmentSlot slot) {
		return this.backingEntity.hasStackEquipped( net.minecraft.entity.EquipmentSlot.valueOf( slot.name() ) );
	}

	@Override
	public ItemStack getStackInSlot(EquipmentSlot slot) {
		return new ForgeItemStack( this.backingEntity.getEquippedStack( net.minecraft.entity.EquipmentSlot.valueOf( slot.name() ) ) );
	}

	@Override
	public void setStackInSlot(EquipmentSlot slot, ItemStack stack) {
		this.backingEntity.equipStack( net.minecraft.entity.EquipmentSlot.valueOf( slot.name() ), ( (ForgeItemStack) stack ).getStack() );
	}
}

File: loadercomplex-forge/src/ver182/kotlin/com/enderzombi102/loadercomplex/forge182/impl/entity/ForgeLivingEntity.kt
package com.enderzombi102.loadercomplex.forge182.impl.entity

import com.enderzombi102.loadercomplex.api.minecraft.entity.PlayerEntity
import com.enderzombi102.loadercomplex.api.minecraft.item.EquipmentSlot
import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack
import com.enderzombi102.loadercomplex.forge182.impl.item.ForgeItemStack
import net.minecraft.entity.LivingEntity
import net.minecraft.entity.player.PlayerEntity as McPlayerEntity
import net.minecraft.util.Hand

open class ForgeLivingEntity(private val backingEntity: LivingEntity) : ForgeEntity(
	backingEntity
),
	com.enderzombi102.loadercomplex.api.minecraft.entity.LivingEntity {
	override fun getHealth(): Float {
		return backingEntity.health
	}

	override fun setHealth(health: Float) {
		backingEntity.health = health
	}

	override fun isPlayer(): Boolean {
		return backingEntity is PlayerEntity
	}

	override fun asPlayer(): PlayerEntity {
		return ForgePlayer(backingEntity as McPlayerEntity)
	}

	override fun isChild(): Boolean {
		return backingEntity.isBaby
	}

	override fun isWaterMob(): Boolean {
		return backingEntity.canBreatheInWater()
	}

	override fun isClimbing(): Boolean {
		return backingEntity.isClimbing
	}

	override fun getArmorProtection(): Int {
		return backingEntity.armor
	}

	override fun getItemInMainHand(): ItemStack {
		return ForgeItemStack(backingEntity.getStackInHand(Hand.MAIN_HAND))
	}

	override fun getItemInOffHand(): ItemStack {
		return ForgeItemStack(backingEntity.getStackInHand(Hand.OFF_HAND))
	}

	override fun hasEquipment(slot: EquipmentSlot): Boolean {
		return backingEntity.hasStackEquipped(net.minecraft.entity.EquipmentSlot.valueOf(slot.name))
	}

	override fun getEquipment(slot: EquipmentSlot): ItemStack {
		return ForgeItemStack(backingEntity.getEquippedStack(net.minecraft.entity.EquipmentSlot.valueOf(slot.name)))
	}

	override fun setEquipment(slot: EquipmentSlot, stack: ItemStack) {
		backingEntity.equipStack(net.minecraft.entity.EquipmentSlot.valueOf(slot.name),
			(stack as ForgeItemStack).stack as net.minecraft.item.ItemStack?
		)
	}
}
